woocommerce/woocommerce-android
# Core support for creating order with coupon

Closes: #9003 
Closes: #9027

Should be reviewed together with (after) https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2728. 

Before merging this PR remember to update FluxC version after the corresponding PR is merged ⚠️ 


### Description

This PR aims to match iOS implementation and adds core support for creating orders with coupons.
* Added new button "+ Add coupon" redirecting to the new coupon entry form
* Updated OrderSyncStrategy classes to take coupon line changes into consideration
* Updated data model and passed couponLines argument to REST client

### Testing instructions

Verify it's possible to apply a coupon when creating an order. Focus on checking if the order gets properly updated after adding/removing coupon to the order.

### Images/gif


https://github.com/woocommerce/woocommerce-android/assets/4527432/8ee19dbf-b327-4e50-bb06-c5656be74ebf

### Next steps
In the next PR, we'll add a local validation of the coupon.
